src/app/components/Account/AccountLink.tsx

```diff
@@ -14,34 +14,69 @@ import { HighlightedText } from '../HighlightedText'
 import { AdaptiveHighlightedText } from '../HighlightedText/AdaptiveHighlightedText'
 import { AdaptiveTrimmer } from '../AdaptiveTrimmer/AdaptiveTrimmer'
 import { AccountMetadataSourceIndicator } from './AccountMetadataSourceIndicator'
+import { useHighlighting } from '../HighlightingContext'
+import { COLORS } from '../../../styles/theme/colors'
 
-const WithTypographyAndLink: FC<{
-  to: string
-  mobile?: boolean
-  children: ReactNode
-  labelOnly?: boolean
-}> = ({ children, to, mobile, labelOnly }) => {
+const WithHighlighting: FC<{ children: ReactNode; address: string }> = ({ children, address }) => {
+  const { highlightedAddress, highlightAddress, releaseAddress } = useHighlighting()
+  useEffect(() => () => releaseAddress(address)) // Release address on umount
+  const isHighlighted = !!highlightedAddress && highlightedAddress.toLowerCase() === address.toLowerCase()
   return (
-    <Typography
-      variant="mono"
-      component="span"
+    <Box
+      onMouseEnter={() => highlightAddress(address)}
+      onMouseLeave={() => releaseAddress(address)}
       sx={{
-        ...(mobile
+        display: 'inline-flex',
+        alignItems: 'center',
+        verticalAlign: 'middle',
+        padding: '4px 4px 2px 4px',
+        ...(isHighlighted
           ? {
-              maxWidth: '100%',
-              overflow: 'hidden',
+              background: COLORS.warningLight,
+              border: `1px dashed ${COLORS.warningColor}`,
+              borderRadius: '6px',
             }
-          : {}),
+          : {
+              margin: '1px',
```

Comment:
`border: 1px dashed transparent` would be more clear that this line attempts to reserve the space for border to avoid resizing

Reply:
Yes, that was the goal. Updated accordingly.